hub/core/chunk_engine/read.py

```diff
@@ -62,18 +62,18 @@ def sample_from_index_entry(
     """Get the unchunked sample from a single `index_map` entry."""
 
     b = bytearray()
-    for chunk_name in index_entry["chunk_names"]:
+    for chunk_name in index_entry[1]:
         chunk_key = os.path.join(key, "chunks", chunk_name)
         last_b_len = len(b)
         b.extend(storage[chunk_key])
 
-    start_byte = index_entry["start_byte"]
-    end_byte = last_b_len + index_entry["end_byte"]
+    start_byte = index_entry[3]
+    end_byte = last_b_len + index_entry[5]
```

Comment:
I _really_ don't like this syntax. We'll have to constantly remember the numbers corresponding to each field, and have to update _everything_ if we ever change the order.

Also, python dicts are guaranteed to be ordered based on _insertion_, so it'll also be tricky to ensure that this ordering is always maintained. 

What's the motivation for doing this conversion? If it's really essential, we should add a layer of abstraction here to interface with the index mapping more cleanly

Reply:
yeah, maybe we should use a class and have it be responsible for the encoding/decoding

Reply:
same thing with the meta, @imshashank recommended this quite some time ago, with accessing meta values like `tensor_meta.min_shape`, i think this is good timing to implement.